src/realm.h

```diff
@@ -1606,14 +1595,24 @@ RLM_API realm_class_key_t realm_object_get_table(const realm_object_t* object);
  */
 RLM_API realm_link_t realm_object_as_link(const realm_object_t* object);
 
+/**
+ * Helper method for making it easier to to convert SDK input to the underlying
+ * `realm_key_path_array_t`.
+ *
+ * @return A pointer to the converted key path array. NULL in case of an error.
+ */
+RLM_API realm_key_path_array_t* realm_create_key_path_array(const realm_t* realm,
+                                                            const realm_class_key_t object_class_key,
+                                                            int user_key_paths_count, const char** user_key_paths);
```

Comment:
I guess `user_key_paths_count` cannot be negative.

Reply:
Sure not.